examples/pennylane_run_qaoa_maxcut.py

```diff
@@ -0,0 +1,291 @@
+# computational basis.
+# In the case of the graph shown above, we want to measure either 0101 or 1010 from our state.
+#
+# .. figure:: ../../examples/figures/qaoa_optimal_state.png
+#   :align: center
+#   :scale: 60%
+#   :alt: optimal_state
+#
+#
+# Implementing QAOA in PennyLane
+# ------------------------------
+#
+# Imports and setup
+# ~~~~~~~~~~~~~~~~~
+#
+# To get started, we import PennyLane along with the PennyLane-provided
+# version of NumPy.
+
+
+import pennylane as qml
+from pennylane import numpy as np
+
+
+##############################################################################
+# Operators
+# ~~~~~~~~~
+# We specify the number of qubits (vertices) with ``n_wires`` and
+# compose the unitary operators using the definitions
+# above. :math:`U_B` operators act on each of the wires, while :math:`U_C`
+# operators act on wires whose corresponding vertices are joined by an edge in
+# the graph. We also define the graph using
+# the list ``graph``, which contains the tuples of vertices defining
+# each edge in the graph.
+
+n_wires = 4
+graph = [(0, 1), (0, 3), (1, 2), (2, 3)]
+
+# unitary operator U_B with parameter beta
+def U_B(beta):
+    for wire in range(n_wires):
+        qml.RX(2 * beta, wires=wire)
+
+
+# unitary operator U_C with parameter gamma
+def U_C(gamma):
+    for edge in graph:
+        wire1 = edge[0]
+        wire2 = edge[1]
+        qml.CNOT(wires=[wire1, wire2])
+        qml.RZ(gamma, wires=wire2)
+        qml.CNOT(wires=[wire1, wire2])
+
+
+##############################################################################
+# With some foresight, we notice that we will need a way to sample
+# a measurement of multiple qubits in the computational basis, so we define
+# a Hermitian operator to do this. The eigenvalues of the operator are
+# the joint qubit measurement values in integer form.
+
+
+def comp_basis_measurement(wires):
+    n_wires = len(wires)
+    return qml.Hermitian(np.diag(range(2 ** n_wires)), wires=wires)
+
+
+##############################################################################
+# Circuit
+# ~~~~~~~
+# Next, we create a quantum device with 4 qubits.
+
+dev1 = qml.device("default.qubit", wires=n_wires, analytic=True, shots=1)
+
+##############################################################################
+# We also require a quantum node which will apply the operators according to the
+# angle parameters, and return the expectation value of the observable
+# :math:`\sigma_z^{j}\sigma_z^{k}` to be used in each term of the objective function later on. The
+# argument ``edge`` specifies the chosen edge term in the objective function, :math:`(j,k)`.
+# Once optimized, the same quantum node can be used for sampling an approximately optimal bitstring
+# if executed with the ``edge`` keyword set to None.
+#
+#  We can also adjust the number of layers in the circuit :math:`p` using a keyword, `n_layers`.
```

Comment:
add what you mean by a `layer` here.

Reply:
something like: one layer contains one U_B and U_C. it can be called a layer because it is a repetitive block of gates in the circuit.

Reply:
I mention this in the 2nd to last paragraph of the section "A quantum circuit". Should I still add this here?

Reply:
I would say so. A layer has a particular meaning when it comes to quantum circuits. see
https://pennylane.readthedocs.io/en/latest/concepts/varcirc.html#architectures

Reply:
Okay got it, thank you